massa-execution-worker/src/execution.rs

```diff
@@ -385,13 +398,27 @@ impl ExecutionState {
         &self,
         operation: &OperationType,
         buyer_addr: Address,
+        current_slot: Slot,
     ) -> Result<(), ExecutionError> {
         // process roll buy operations only
         let roll_count = match operation {
             OperationType::RollBuy { roll_count } => roll_count,
             _ => panic!("unexpected operation type"),
         };
 
+        // control vesting max_rolls for buyer address
+        if let Some(vesting_range) = self.find_vesting_range(&buyer_addr, &current_slot) {
+            let rolls = self.get_final_and_candidate_rolls(&buyer_addr);
+            // (candidate_rolls + amount to buy)
+            let max_rolls = rolls.1 + roll_count;
```

Comment:
Security issue: roll_count is unverified input and could overflow `rolls.1 + roll_count`

General rule: always think twice before writing an arithmetic operator, especially when it operates on untrusted input